File: scratch/compact-guards-integration-20260916/after/src/Test/PolymorphismFFICheatcode.php
<?php
interface Monoidish {
    public function mempty_();
    public function mappend_($x, $y);
}
class IntMonoidish implements Monoidish {
    public function mempty_() { return 1; }
    public function mappend_($x, $y) { return $x + $y; }
}
$exports['runPolymorphismFFICheatcode'] = function($limit) {
    $n = (int)$limit;
    $acc = 0;
    $m = new IntMonoidish();
    while ($n > 0) {
        $acc = $m->mappend_($acc, $m->mempty_());
        $n--;
    }
    return $acc;
};
return $exports;

File: scratch/gopurs-adt-reuse-validation-20260916/after-pure/src/Test/PolymorphismFFICheatcode.rs
trait Monoidish {
    fn mempty_(&self) -> i64;
    fn mappend_(&self, x: i64, y: i64) -> i64;
}
struct IntMonoidish;
impl Monoidish for IntMonoidish {
    fn mempty_(&self) -> i64 { 1 }
    fn mappend_(&self, x: i64, y: i64) -> i64 { x + y }
}
pub fn Test_PolymorphismFFICheatcode_runPolymorphismFFICheatcode(mut limit: i64) -> i64 {
    let mut acc = 0;
    let m = IntMonoidish;
    while limit > 0 {
        acc = m.mappend_(acc, m.mempty_());
        limit -= 1;
    }
    acc
}
